api/clientcredentialsloginprovider.go

```diff
@@ -32,6 +32,12 @@ type clientCredentialsLoginProvider struct {
 	clientSecret string
 }
 
+// Token implements [LoginProvider.Token]
+// Returning the client secret used for logging in to a Juju controller.
+func (p *clientCredentialsLoginProvider) Token() (string, error) {
+	return p.clientSecret, nil
```

Comment:
Is clientSecret guaranteed to be not ""?
If not we'd need to return ErrorTokenNotAvailable

We have `NewClientCredentialsLoginProvider` above but there's no check to validate clientSecet (nor clientID).

Reply:
With the introduction of the error, the clientSecret could be "". There is not necessarily a problem with that, it's just based on whatever was provided. I don't think there is a meaningful check we could do in `NewClientCredentialsLoginProvider`, possibly we could prevent the user from providing an empty value if we wanted, I'm open to either way.

Reply:
To clarify my previous statement I've bolded the bits that I think are relevant,
> With the introduction of the error, the clientSecret **could be** "".

What I was trying to say there is that with the use of the error, the clientSecret can be an empty string if the provider deems that to be a valid value. Currently, the only time an error is returned is when a provider cannot provide a token until its `Login` method has been run - this applies to the sessionToken provider. The remaining providers are expected to return whatever string was used during creation of the provider, empty or not.

>If we want the semantics to be that we never return an empty value for the token

We no longer want to do that.

If you think we _should_ do that, that can be figured out. I'd also like to hear @tlm's thoughts on this.

Reply:
The Token() method should either always return a `ErrorTokenNotAvailable` error when the result string is "", or not. The API contract for an interface method must be implemented consistently across all implementations.